crates/host/src/policy.rs

```diff
@@ -10,110 +10,145 @@ use futures::{
-    /// The contract ID of the provider, or None if the target is an actor
-    #[serde(rename = "contractId")]
-    pub contract_id: Option<String>,
-    /// The link name of the provider, or None if the target is an actor
-    #[serde(rename = "linkName")]
-    pub link_name: Option<String>,
+/// Relevant policy information for evaluating a component
+pub struct ComponentInformation {
+    /// The unique identifier of the component
+    #[serde(rename = "componentId")]
+    pub component_id: String,
+    /// The image reference of the component
+    #[serde(rename = "imageRef")]
+    pub image_ref: String,
+    /// The requested maximum number of concurrent instances for this component
+    #[serde(rename = "maxInstances")]
+    pub max_instances: u32,
+    /// Annotations associated with the component
+    pub annotations: BTreeMap<String, String>,
+    /// Claims, if embedded, within the component
+    pub claims: Option<PolicyClaims>,
+}
+
+#[derive(Clone, Debug, Default, Eq, PartialEq, Serialize, Hash)]
+/// Relevant policy information for evaluating a provider
+pub struct ProviderInformation {
+    /// The unique identifier of the provider
+    #[serde(rename = "providerId")]
+    pub provider_id: String,
+    /// The image reference of the provider
+    #[serde(rename = "imageRef")]
+    pub image_ref: String,
+    /// Annotations associated with the provider
+    pub annotations: BTreeMap<String, String>,
+    /// Claims, if embedded, within the provider
+    pub claims: Option<PolicyClaims>,
+}
+
+#[derive(Clone, Debug, Default, Eq, PartialEq, Serialize, Hash)]
+/// A request to invoke a component function
+pub struct PerformInvocationRequest {
+    /// The interface of the invocation
+    pub interface: String,
+    /// The function of the invocation
+    pub function: String,
+    /// Target of the invocation
+    pub target: ComponentInformation,
 }
 
 /// Relevant information about the host that is receiving the invocation, or starting the actor or provider
 #[derive(Clone, Debug, Serialize)]
 pub struct HostInfo {
-    /// The public key of the host
+    /// The public key ID of the host
     #[serde(rename = "publicKey")]
     pub public_key: String,
-    /// The ID of the lattice the host is running in
-    #[serde(rename = "latticeId")]
-    pub lattice_id: String,
+    /// The name of the lattice the host is running in
+    #[serde(rename = "lattice")]
+    pub lattice: String,
     /// The labels associated with the host
     pub labels: HashMap<String, String>,
-    /// The host's list of issuers it will accept invocations from
-    #[serde(rename = "clusterIssuers")]
-    pub cluster_issuers: Vec<String>,
 }
 
 /// The action being requested
-#[derive(Clone, Debug, Eq, PartialEq, Serialize, Hash)]
-pub enum Action {
-    /// The host is checking whether it may invoke the target actor
-    #[serde(rename = "perform_invocation")]
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Serialize, Hash)]
+pub enum RequestKind {
```

Comment:
Because a policy service can be written by anyone, should this be `#[serde(untagged)]` so it has some cleaner JSON?

Reply:
I don't know if it needs to be, the request format looks like this:
```json
{
  "requestId": "018dfb36-c639-ab3b-caa3-f7823e8d04c9",
  "kind": "startComponent",
  "version": "v1",
...
}
```

I know untagging the request body prevented a weird nesting, but here since it's just a String rename I think it's okay

Reply:
Ok, that should be fine then